bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniPatientServiceImpl.java

```diff
@@ -93,4 +99,30 @@ public PatientConfigResponse getConfig() {
         return patientDao.getByAIsToB(aIsToB);
     }
 
+    @Override
+    public List<PatientResponse> luceneHibernateSearch(PatientSearchParameters searchParameters) {
+        List<PatientResponse> luceneHibernateResponse = new ArrayList<>();
+        if(StringUtils.isNotEmpty(searchParameters.getIdentifier()) || StringUtils.isNotEmpty(searchParameters.getName()) || StringUtils.isNotEmpty(searchParameters.getCustomAttribute())){
+            List<PatientResponse> luceneResponse = luceneSearch(searchParameters);
+
+            for(PatientResponse patientResponse: luceneResponse){
+                Patient patient = patientService.getPatient(patientResponse.getPersonId());
+                searchParameters.setIdentifier(patient.getPatientIdentifier().getIdentifier());
+                if(StringUtils.isNotEmpty(searchParameters.getName())){
+                    searchParameters.setName("");
+                }
+                if( StringUtils.isNotEmpty(searchParameters.getCustomAttribute())){
+                    searchParameters.setCustomAttribute("");
+                }
+                luceneHibernateResponse.addAll(search(searchParameters));
```

Comment:
You're firing a hibernate/sql search _for every_ result that comes back from the lucene search? Are we guaranteed that the number of these searches will be small?
Can't we just do one lucene search, one hibernate search, and intersect them? (Or even include the lucene responses in a "patientId in :list" in the hibernate/sql query?
(And, have we verified that actually using lucene to search for the other parameters is that hard to do?)